src/main/kotlin/org/jetbrains/research/libsl/nodes/references/TypeReference.kt

```diff
@@ -3,11 +3,11 @@ package org.jetbrains.research.libsl.nodes.references
 import org.jetbrains.research.libsl.context.LslContextBase
 import org.jetbrains.research.libsl.type.*
 
-open class TypeReference(
-    val name: String,
-    val isPointer: Boolean,
-    var typeBound: GenericTypeBound = GenericTypeBound.EMPTY,
-    val genericReferences: MutableList<TypeReference>,
+abstract class TypeReference(
+    open val name: String,
+    open val isPointer: Boolean = false,
+    open var typeBound: GenericTypeBound = GenericTypeBound.EMPTY,
+    open val genericReferences: MutableList<TypeReference> = mutableListOf(),
```

Comment:
Why the `UnionExpressionTypeReference` for example should have fields associated with generic types (i.e. `typeBound` and `genericReferences`)?

Reply:
Fixed; TypeReference fields were separated to classes where they are using;